django/forms/fields.py

```diff
@@ -1211,3 +1212,65 @@ def to_python(self, value):
             except ValueError:
                 raise ValidationError(self.error_messages['invalid'], code='invalid')
         return value
+
+
+class InvalidJSONInput(str):
+    pass
+
+
+class JSONString(str):
+    pass
+
+
+class JSONField(CharField):
+    default_error_messages = {
+        'invalid': _('Enter a valid JSON value.'),
+    }
+    widget = Textarea
+
+    def __init__(self, encoder=None, decoder=None, **kwargs):
+        self.encoder, self.decoder = encoder, decoder
```

Comment:
Nitpick : can we remove the parallel assignment here? I have a draft blog post on why it's slower (tuple construction just to immediately deconstruct), and less clear (if an exception is raised, unknown which variable caused it)